Sources/ArcGISToolkit/Components/OfflineMapAreasView/OfflineMapAreasView.MapViewModel.swift

```diff
@@ -0,0 +1,168 @@
+        /// The portal item ID of the web map.
+        private var portalItemID: String?
+        
+        /// The offline map of the downloaded preplanned map area.
+        private var offlineMap: Map?
+        
+        /// The offline map task.
+        let offlineMapTask: OfflineMapTask
+        
+        /// The url for the preplanned map areas directory.
+        let preplannedDirectory: URL
+        
+        /// The mobile map packages created from mmpk files in the documents directory.
+        @Published private(set) var mobileMapPackages: [MobileMapPackage] = []
+        
+        /// The preplanned offline map information.
+        @Published private(set) var preplannedMapModels: Result<[PreplannedMapModel], Error>?
+        
+        /// A Boolean value indicating whether the map has preplanned map areas.
+        @Published private(set) var hasPreplannedMapAreas = false
+        
+        /// A Boolean value indicating whether the user has authorized notifications to be shown.
+        var canShowNotifications = false
+        
+        /// The job manager.
+        let jobManager = JobManager.shared
+        
+        init(map: Map) {
+            offlineMapTask = OfflineMapTask(onlineMap: map)
+            
+            if let portalItemID = map.item?.id?.rawValue {
+                FileManager.default.createDirectories(for: portalItemID)
+                
+                preplannedDirectory = FileManager.default.preplannedDirectory(portalItemID: portalItemID)
+            } else {
+                preplannedDirectory = FileManager.default.documentsDirectory
+            }
+        }
+        
+        /// Gets the preplanned map areas from the offline map task and creates the
+        /// offline map models.
+        func makePreplannedOfflineMapModels() async {
+            preplannedMapModels = await Result {
+                try await offlineMapTask.preplannedMapAreas
+                    .sorted(using: KeyPathComparator(\.portalItem.title))
+                    .compactMap {
+                        PreplannedMapModel(
+                            preplannedMapArea: $0,
+                            offlineMapTask: self.offlineMapTask,
+                            preplannedDirectory: self.preplannedDirectory
+                        )
+                    }
+            }
+            if let models = try? preplannedMapModels?.get() {
+                hasPreplannedMapAreas = !models.isEmpty
+            }
+        }
+        
+        /// Loads locally stored mobile map packages for the map's preplanned map areas.
+        func loadPreplannedMobileMapPackages() {
+            // Create mobile map packages with saved mmpk files.
+            let mmpkFiles = searchFiles(in: preplannedDirectory, with: "mmpk")
+            
+            self.mobileMapPackages = mmpkFiles.map(MobileMapPackage.init(fileURL:))
+            
+            // Pass mobile map packages to preplanned map models.
```

Comment:
I think it might make more sense to pass the portal item ID to the preplanned map model and have it check the directory for a matching mmpk. I think it could be that this code is a parent object doing too much on behalf of a child object.

Reply:
The parent object checks for a matching mmpk to avoid passing a copy of the parent's mobile map packages array to each child object. I can make this change if preferred.

Reply:
Yes the preplanned map model can look up the mmpk files. I will move the check into the preplanned model.